src/Auth/Models/Companies.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Canvas\Auth\Models;
+
+use Baka\Database\Model;
+use Exception;
+use Phalcon\Validation;
+use Phalcon\Validation\Validator\PresenceOf;
+
+class Companies extends Model
+{
+    const DEFAULT_COMPANY = 'DefaulCompany';
+
+    public string $name;
+    public ?string $profile_image = null;
+    public ?string $website = null;
+    public int $users_id;
+
+    /**
+     * Initialize method for model.
+     */
+    public function initialize()
+    {
+        $this->setSource('companies');
+        $this->belongsTo('users_id', 'Baka\Auth\Models\Users', 'id', ['alias' => 'user']);
+        $this->hasMany('id', 'Baka\Auth\Models\CompanySettings', 'id', ['alias' => 'settings']);
+    }
+
+    /**
+     * Model validation.
+     *
+     * @return void
+     */
+    public function validation()
+    {
+        $validator = new Validation();
+
+        $validator->add(
+            'name',
+            new PresenceOf([
+                'model' => $this,
+                'required' => true,
+            ])
+        );
+
+        return $this->validate($validator);
+    }
+
+    /**
+     * Register a company given a user and name.
+     *
+     * @param  Users  $user
+     * @param  string $name
+     *
+     * @return Companies
+     */
+    public static function register(Users $user, string $name) : Companies
+    {
+        $company = new self();
+        $company->name = $name;
+        $company->users_id = $user->getId();
+
+        if (!$company->save()) {
+            throw new Exception(current($company->getMessages()));
+        }
+
+        return $company;
+    }
+
+    /**
+     * After creating the company.
+     *
+     * @return void
+     */
+    public function afterCreate()
```

Comment:
@rwhite27 dont thing this is needed anymore since on the other class this is overwriten